Sources/StreamChatUI/ChatMessageList/ChatThreadVC.swift

```diff
@@ -428,3 +442,38 @@ open class _ChatThreadVC<ExtraData: ExtraDataTypes>:
         print(#function, indexPath)
     }
 }
+
+extension _ChatThreadVC {
+    func reply(at indexPath: IndexPath) -> _ChatMessage<ExtraData> {
+        messageController.replies[indexPath.item]
+    }
+
+    func attachmentViewInjectorClass(for message: _ChatMessage<ExtraData>) -> _AttachmentViewInjector<ExtraData>.Type? {
+        let attachmentCounts = message.attachmentCounts
+
+        if attachmentCounts.keys.contains(.image) {
+            return components.galleryAttachmentInjector
+        } else if attachmentCounts.keys.contains(.giphy) {
+            return components.giphyAttachmentInjector
+        } else if attachmentCounts.keys.contains(.file) {
+            return components.filesAttachmentInjector
+        } else if attachmentCounts.keys.contains(.linkPreview) {
+            return components.linkAttachmentInjector
+        } else {
+            return nil
+        }
```

Comment:
Can we move this into a similar component to `ChatMessageLayoutOptionsResolver` and share it between the two VCs?

Reply:
There's a lot of duplicate logic right now in both ChatThreadVC and ChatMessageList can we do this refactor in this ticket? https://stream-io.atlassian.net/secure/RapidBoard.jspa?rapidView=52&projectKey=CIS&selectedIssue=CIS-860